Automattic/jetpack
# Update query hooks and introduce window state getters

## Proposed changes:
* Redesign parameters for query hooks
* Remove `explicitKey` param
* Add My Jetpack window state getter

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:
Basically make sure that My Jetpack still works and loads correctly. You can run the following test:

1. Checkout this branch via your local dev environment
2. Make sure your site is user connected
3. In `projects/packages/my-jetpack/src/class-rest-purchases.php` add the following line to the `get_site_current_purchases` function to return an error early
```php
return new \WP_Error( 'site_data_fetch_failed', 'Site data fetch failed', array( 'status' => $response_code ? $response_code : 400 ) );
```
4. Make sure the purchases error shows up as it should
![image](https://github.com/Automattic/jetpack/assets/65001528/06cc8a70-4d4b-492c-b5cc-49747855789d)
5. Now remove that error line and head to `projects/packages/my-jetpack/src/class-rest-zendesk-chat.php` and return an error early in the `get_chat_authentication` function
```php
		return new \WP_Error( 'chat_authentication_failed', 'Chat authentication failed', array( 'status' => $response_code ) );
```
6. Make sure the API call fails and that no error shows
![image](https://github.com/Automattic/jetpack/assets/65001528/98e3a0c7-d94c-425a-b0df-be0e04ff7d6e)
